gio-subclass/src/application.rs

```diff
@@ -29,7 +29,20 @@ impl ArgumentList{
         }
     }
 
-    fn remove(){
+    // remove the item at index `idx` and shift the raw array
+    pub fn remove(&mut self, idx: usize){
+        self.items.remove(idx);
+
+        unsafe{
+            let n_args = glib_ffi::g_strv_length(*self.ptr);
+            assert!((idx as u32) < n_args);
+            glib_ffi::g_strfreev(*self.ptr.offset(idx as isize));
```

Comment:
This is incorrect. You need to call `glib_ffi::g_free` on the item to be removed